src/ImageSharp/PixelFormats/PixelBlenders/PorterDuffFunctions.cs

```diff
@@ -21,6 +21,12 @@ namespace SixLabors.ImageSharp.PixelFormats.PixelBlenders;
 /// </remarks>
 internal static partial class PorterDuffFunctions
 {
+    private const int BlendAlphaControl = 0b_10_00_10_00;
+    private const int ShuffleAlphaControl = 0b_11_11_11_11;
+    private static readonly Vector256<float> Vector256Half = Vector256.Create(0.5F);
+    private static readonly Vector256<float> Vector256One = Vector256.Create(1F);
+    private static readonly Vector256<float> Vector256Two = Vector256.Create(2F);
```

Comment:
This is going to be worse than direct usage for .NET 5+

Starting in .NET 5 there was some special support added for `Vector###.Create(cns)` and `Vector###.Create(cns, ..., cns)`. These APIs will now generate a "method local constant" which makes it quite a bit more efficient than even a `static readonly` will be.

In .NET 5 this support only existed in the late phases of the JIT (lowering). That support was improved a bit more in .NET 6, 7, and now in 8 as well. Starting in 7 in particular we now have a direct node (`GT_CNS_VEC`) which allows other phases of the JIT to take advantage of this. In .NET 8 we've started adding constant folding support as well.

Reply:
If you want to centralize such "constants", I'd recommend returning them from a property instead:
```csharp
private static Vector256<float> Vector256Half => Vector256.Create(0.5F);
private static Vector256<float> Vector256One => Vector256.Create(1F);
private static Vector256<float> Vector256Two => Vector256.Create(2F);
```

Reply:
Oh that's very interesting. Will revert and look into other examples in the codebase.

What about multiple references to a local variable within a single method? Should I inline them?

```c#
public static Vector256<float> Screen(Vector256<float> backdrop, Vector256<float> source)
{
    Vector256<float> vOne = Vector256.Create(1F);
    return Avx.Subtract(vOne, Avx.Multiply(Avx.Subtract(vOne, backdrop), Avx.Subtract(vOne, source)));
}
```

Reply:
Not sure off the top of my head. In .NET 7/8 I'd expect us to be doing the right thing and recognizing them as a "common subexpression". I'd also expect the same for .NET 6, but the support wasn't quite as good and so manually hoisting it to a local may provide better results and shouldn't hurt on any of the target frameworks.

In general you can assume that most `Create(cns)` and `Create(cns, ..., cns)` come from memory and so using a local can help ensure there is only one memory access. The special consideration is `Zero` and `AllBitsSet` which are generated dynamically using a "zero cost" instruction instead.